code_review/dp/code_review_daw_parms.Rmd

```diff
@@ -0,0 +1,280 @@
+                        data=luna_final,
+                        family="binomial",
+                        glmerControl(optimizer = "bobyqa"))
+summary(intmodeltoplot)
+
+#Marginal effect: Common vs Rare
+cr_me <- ggpredict(intmodeltoplot, terms = "commonraredummy")
+ggplot(cr_me, 
+       aes(x = as.factor(x), 
+           y = predicted)) + 
+  geom_point(size = 3) + 
+  geom_errorbar(aes(ymin = conf.low, 
+                    ymax = conf.high), 
+                width = 0.2) +
+  labs(title = "Main effect of tranistion type",
+    x = "Transition Type",
+    y = "Probability of FSS") +
+   scale_y_continuous(labels = scales::percent) +
+  theme_modern()
+
+#Marginal effect: Reward
+rew_me <- ggpredict(intmodeltoplot, terms = "moneylagdummy")
+ggplot(rew_me, 
+       aes(x = as.factor(x), 
+           y = predicted)) + 
+  geom_point(size = 3) + 
+  geom_errorbar(aes(ymin = conf.low, 
+                    ymax = conf.high), 
+                width = 0.2) +
+  labs(title = "Main effect of reward",
+    x = "Rewarded",
+    y = "Probability of FSS") +
+   scale_y_continuous(labels = scales::percent) +
+  theme_modern()
+
+#Marginal effect: Reward x Transition Type
+cr_rew_me <- ggpredict(intmodeltoplot, terms = c("moneylagdummy", "commonraredummy"))
+ggplot(cr_rew_me, 
+       aes(x = as.factor(x), 
+           y = predicted,
+           group = group,
+           colour = group)) + 
+  geom_point(size = 3) + 
+  geom_errorbar(aes(ymin = conf.low, 
+                    ymax = conf.high), 
+                width = 0.2) +
+  labs(title = "Transition type x reward interaction",
+    x = "Reward",
+    y = "Probability of FSS") +
+   scale_y_continuous(labels = scales::percent) +
+  scale_color_discrete(name = "Transition Type") +
+  theme_modern()
+```
+
+Fun!
+
+
+Extract fixed and random effects. Then, add fixed effects to random effects to create MB/MF/FSS scores
+
+
+```{r Extract Fixed and Random Effects}
+#Extracting random effects
+dawranefint <- ranef(intmodeltoplot)
+head(dawranefint$`id:visitnum`)
+
+#Extracting fixed effects
+dawfixedeffectsint <- summary(intmodeltoplot)$coefficients
+dawfixedeffectsint
+
+#Creating dataframe consisting of MB/MF/Habit parameter.
+dawsubjecterms <- data.frame(
+  idtemp = row.names(dawranefint$`id:visitnum`), 
+  modelbased = dawranefint$`id:visitnum`$`commonraredummy:moneylagdummy` + 
+    dawfixedeffectsint[4,1], 
+  modelfree = dawranefint$`id:visitnum`$moneylagdummy + 
+    dawfixedeffectsint[3,1],
+  commonrare = dawranefint$`id:visitnum`$commonraredummy + 
+    dawfixedeffectsint[2,1],
+  firststagestay = dawranefint$`id:visitnum`$`(Intercept)` + 
+    dawfixedeffectsint[1,1])
+  
+#Create id column
+dawsubjecterms$id <- unlist(lapply(strsplit(as.character(dawsubjecterms$idtemp),
+                                            ":"),'[',1))
+#Create visitnum column
+dawsubjecterms$visitnum <- unlist(lapply(strsplit(as.character(dawsubjecterms$idtemp),
```

Comment:
[`stringr::str_split`](https://stringr.tidyverse.org/reference/str_split.html) is marginally nicer